client/src/components/Applications/AppForm/AppForm.js

```diff
@@ -550,13 +554,11 @@ export const AppSchemaForm = ({ app }) => {
                     />
                   ) : null}
                 </div>
-                <Button type="submit" color="primary">
+                <Button type="submit" color="primary" disabled={!isValid}>
                   {jobSubmission.submitting && (
                     <LoadingSpinner placement="inline" />
                   )}{' '}
-                  {(Object.keys(errors).length || jobSubmission.error) && (
```

Comment:
r.e. the removal of the check on the `errors` object, did you find that it was not in use?

Reply:
Ah, is it that we don't want to show the warning symbol before they've had a chance to alter the form submission?

Reply:
yes, exactly.  Submit button is disabled if there is a validation error (now using `isValid` for that check instead of `errors`).